### PR TITLE
[AzCopyV10][StgExp][Github971] Cannot scan public folder with /* when performing copy

### PR DESCRIPTION
**Pull request by mohsha-msft**
Issue:
https://github.com/Azure/azure-storage-azcopy/issues/971
https://github.com/microsoft/AzureStorageExplorer/issues/3394

Unable to scan virtual directories of a public container.

### CODE REVIEW

File: cmd/credentialUtil.go [hunk 1]
```diff
@@ -125,6 +126,18 @@ func getBlobCredentialType(ctx context.Context, blobResourceURL string, canBePub
 				isPublicResource = true
 			}
 		} else {
+			// Scenario 2: When resourceURL points to a virtual directory.
+			// Check if the virtual directory is accessible by doing GetProperties on container.
+			// Virtual directory can be accessed/scanned only when its parent container is public.
+			bURLParts := azblob.NewBlobURLParts(*resourceURL)
+			bURLParts.BlobName = ""
+			containerURL := azblob.NewContainerURL(bURLParts.URL(), p)
+			if _, err := containerURL.GetProperties(ctx, azblob.LeaseAccessConditions{}); err == nil {
```
Comment (nakulkar-msft) [inline cmd/credentialUtil.go:135]: This statement is repeated in both the blocks. Scenario1 and scenario2 are the same, and should be merged.  A virtual dir can be public if container is public.

Comment (mohsha-msft) [reply cmd/credentialUtil.go]: Done. I concur that code duplicity can be reduced here. I've extracted out the common part of both if...else flow and for scenario 3 only I'm calling blob.GetProperties().

File: cmd/copy.go [hunk 2]
```diff
@@ -242,7 +242,7 @@ func (raw rawCopyCmdArgs) cookWithId(jobId common.JobID) (cookedCopyCmdArgs, err
 	tempDest := raw.dst
 
 	if strings.EqualFold(tempDest, common.Dev_Null) && runtime.GOOS == "windows" {
-		tempDest = common.Dev_Null // map all capitalizations of "NUL"/"nul" to one because (on Windows) they all mean the same thing
+		tempDest = common.Dev_Null // map all capitalization of "NUL"/"nul" to one because (on Windows) they all mean the same thing
```
Comment (adreed-msft) [inline cmd/copy.go:245]: "capitalizations" was correct here. On a side note, perhaps this is not the PR to run around correcting comment grammar in unrelated files... Though, we should certainly go through the code and ensure all comments make sense at some point.

**Review by adreed-msft — APPROVED** [review]

**Comment by adreed-msft** [comment]
Nice job, this closely matches the semantics of the service side (where it's either the entire container, or just the blobs that are public).

File: cmd/copy.go (hunk 2 above)
Comment (nakulkar-msft) [reply cmd/copy.go:245]: Adding to adreed's comment, changing the comments will alter the change description of the line. Change descriptions are helpful to know the history and reason for a change. I'd prefer if this file were unchanged.

**Review by nakulkar-msft — APPROVED** [review]
